extensions/wikia/AdEngine/tests/AdEngine2ContextServiceTest.php

```diff
@@ -130,74 +131,102 @@ public function adContextDataProvider() {
-				'expectedTargeting' => ['pageIsHub' => true, 'wikiIsCorporate' => true]
+				'expectedTargeting' => [
+					'newWikiCategories' => [ 'test', 'test' ],
+					'pageIsHub' => true,
+					'wikiIsCorporate' => true
+				]
 			],
 			[
 				'titleMockType' => 'article',
 				'flags' => ['wgWikiDirectedAtChildrenByFounder'],
 				'expectedOpts' => [],
-				'expectedTargeting' => ['wikiDirectedAtChildren' => true]
+				'expectedTargeting' => [ 'newWikiCategories' => [ 'test', 'test' ], 'wikiDirectedAtChildren' => true ]
 			],
 			[
 				'titleMockType' => 'article',
 				'flags' => ['wgWikiDirectedAtChildrenByStaff'],
 				'expectedOpts' => [],
-				'expectedTargeting' => ['wikiDirectedAtChildren' => true]
+				'expectedTargeting' => [ 'newWikiCategories' => [ 'test', 'test' ], 'wikiDirectedAtChildren' => true ]
 			],
 			[
 				'titleMockType' => 'mainpage',
 				'flags' => [],
 				'expectedOpts' => [],
-				'expectedTargeting' => ['pageType' => 'home']
+				'expectedTargeting' => [ 'newWikiCategories' => [ 'test', 'test' ], 'pageType' => 'home' ]
 			],
 			[
 				'titleMockType' => 'search',
 				'flags' => [],
 				'expectedOpts' => ['pageType' => 'search'],
-				'expectedTargeting' => ['pageType' => 'search', 'pageName' => 'Special:Search']
+				'expectedTargeting' => [
+					'newWikiCategories' => [ 'test', 'test' ],
+					'pageType' => 'search',
+					'pageName' => 'Special:Search'
+				]
 			],
 
 			$defaultParameters + ['expectedMappings' => [
-				'sourceVertical' => 'tv',
+				'newVertical' => 'tv',
 				'expectedMappedVertical' => 'ent'
 			]],
 
 			$defaultParameters + ['expectedMappings' => [
-				'sourceVertical' => 'games',
+				'newVertical' => 'games',
 				'expectedMappedVertical' => 'gaming'
 			]],
 
 			$defaultParameters + ['expectedMappings' => [
-				'sourceVertical' => 'books',
+				'newVertical' => 'books',
 				'expectedMappedVertical' => 'ent'
 			]],
 
 			$defaultParameters + ['expectedMappings' => [
-				'sourceVertical' => 'comics',
+				'newVertical' => 'comics',
 				'expectedMappedVertical' => 'ent'
 			]],
 
 			$defaultParameters + ['expectedMappings' => [
-				'sourceVertical' => 'lifestyle',
+				'newVertical' => 'lifestyle',
 				'expectedMappedVertical' => 'life'
 			]],
 
 			$defaultParameters + ['expectedMappings' => [
-				'sourceVertical' => 'not-existing',
+				'newVertical' => 'not-existing',
 				'expectedMappedVertical' => 'error'
 			]],
 
 			$defaultParameters + ['expectedMappings' => [
-				'verticalFromCategoryInfo' => 'Wikia',
-				'sourceVertical' => 'other',
+				'oldVertical' => 'Wikia',
+				'newVertical' => 'other',
 				'expectedMappedVertical' => 'wikia']
 			],
+
+			array_merge($defaultParameters, [
+				'expectedTargeting' => [ 'newWikiCategories' => [ 'test', 'test1' ] ],
+				'categories' => [ 'old' => ['test'], 'new' => ['test1'] ]
+			]),
+
+			array_merge($defaultParameters, [
+				'expectedTargeting' => [ 'newWikiCategories' => [ 'test', 'test', 'test1' ] ],
```

Comment:
Good question! I don't think we've got the same categories names but it won't hurt us if we use `array_unique()` to make sure if we send only unique values to DFP :)